backend/src/main/java/com/bakdata/conquery/io/storage/MetaStorage.java

```diff
@@ -54,41 +56,22 @@ public void openStores(ObjectMapper mapper) {
 	}
 
 	@Override
-	public void loadData() {
-		Preconditions.checkNotNull(authUser, "User storage was not created");
-		Preconditions.checkNotNull(authRole, "Role storage was not created");
-		Preconditions.checkNotNull(authGroup, "Group storage was not created");
-		Preconditions.checkNotNull(executions, "Execution storage was not created");
-		Preconditions.checkNotNull(formConfigs, "FormConfig storage was not created");
+	protected Graph<KeyIncludingStore<?, ?>> getStoreDependencies() {
+		final MutableGraph<KeyIncludingStore<?, ?>> storeOrder = GraphBuilder.directed()
+																			 .allowsSelfLoops(false).build();
 
-		authUser.loadData();
-		authRole.loadData();
-		authGroup.loadData();
+		storeOrder.addNode(authUser);
 
-		executions.loadData();
-		formConfigs.loadData();
-	}
-
-	@Override
-	public void clear() {
-		centralRegistry.clear();
-
-		executions.clear();
-		formConfigs.clear();
+		storeOrder.putEdge(authUser, authRole);
+		storeOrder.putEdge(authRole, authGroup);

```

Comment:
Sollte das nicht umgedreht sein? 
```suggestion
		storeOrder.putEdge(authRole, authUser);
		storeOrder.putEdge(authUser, authGroup);
		storeOrder.putEdge(authRole, authGroup);

```

Also links dependency und rechts dependent

Reply:
Doch, bin selber gerade verwirrt warum ich das so geschrieben habe :D